docs/DebuggerHTML-Components-Reducers-Actions.md

```diff
@@ -0,0 +1,663 @@
+
+
+The App component uses two SplitBox components to separate the
+presentation of the app into three different sections. Two Draggable
+components are used to allow each of the sections to be expanded or
+collapsed.
+
+![](https://docs.google.com/drawings/d/1lAEyyci8SQZzh4Dk-EowX0wGnvyOISGO3sqdtzgQqoo/pub?w=960&h=720)
+[Click here to Edit](https://docs.google.com/drawings/d/1lAEyyci8SQZzh4Dk-EowX0wGnvyOISGO3sqdtzgQqoo/edit?usp=sharing)
+##Source tree view
+
+The left-most section of the application displays the source tree for
+the application being debugged. Three components are used to manage
+the display of this data:
+
+* The Sources component is
+aware of the Redux state and the other components are not — it passes required properties to be rendered down to the other
+two.
+
+* The SourcesTree component is primarily responsible for rendering
+the tree with a set of files/folders that are passed in as a property.
+
+* The ManagedTree component uses React local State to track which nodes
+have been expanded or collapsed and which node or leaf on the tree has
+focus.
+
+The Sources component encapsulates SourcesTree and
+SourcesTree encapsulates ManagedTree.
+
+![](https://docs.google.com/drawings/d/1dOCy4BePfX77ky3yUTlZRnAeeFIIi4UAJcYaYmYvUcY/pub?w=960&h=720)
+[Click here to Edit](https://docs.google.com/drawings/d/1dOCy4BePfX77ky3yUTlZRnAeeFIIi4UAJcYaYmYvUcY/edit?usp=sharing)
+##Source editor/file search
+
+The center portion of the application displays either the source editor or a file search entry box. If the editor is displayed rendering is handled by three main components and one dynamic component.
+
+* At the top of the editor is the SourceTabs component, which is responsible for rendering tabs for every open file and highlighting the tab of the file currently open.
+
+* The Editor component is responsible for rendering the text, gutters and breakpoints for the currently selected file.  Debugger.html uses the CodeMirror npm package to do the actual rendering. The Editor component manages calls to CodeMirror.
+
+* Any time a breakpoint is set, the Editor creates a dynamic component called Breakpoint. The Breakpoint component is contained in the EditorBreakpoint.js file. The BreakPoint component also makes calls to CodeMirror for actual rendering of the breakpoint within the editor.
+
+* The last component on the page is the SourceFooter component. This component renders buttons for blackboxing and prettify source functions.
+
+![](https://docs.google.com/drawings/d/1PC63VABa0x-W3hACi7ASXgawQeowbAvA_aqN_Z1wpRI/pub?w=960&h=720)
+[Click here to Edit](https://docs.google.com/drawings/d/1PC63VABa0x-W3hACi7ASXgawQeowbAvA_aqN_Z1wpRI/edit?usp=sharing)
+
+At any time a user can search the sources for a specific string by
+pressing cmd-&gt;p. This will replace all of the components in the
+center section with a search box. The search box is rendered using the

```

Comment:
these are dynamically created components and I wanted it to be clear that you can create many of them.